### PR TITLE
Feature/21 json mapping repository fix srp

### PR DESCRIPTION
**Pull request by michalrys**
related to https://github.com/EngineCodeDev/ec-commons/issues/21 - see description there.

all tests are passed

### CODE REVIEW

File: src/main/java/dev/enginecode/eccommons/infrastructure/json/repository/mapping/Mapping.java [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package dev.enginecode.eccommons.infrastructure.json.repository.mapping;
+
+public interface Mapping {
```
Comment (tom-zoladz) [inline src/main/java/dev/enginecode/eccommons/infrastructure/json/repository/mapping/Mapping.java:3]: I think the name is too much general as the mappers are not only objects for json serialising/deserialising, but also mappers as for example design pattern to change one object into another. Could we modify it to something more precise, e.g. JsonConverter (JsonMapper is best but used by the implementation)

File: src/main/java/dev/enginecode/eccommons/infrastructure/json/repository/JsonMappingRepository.java [hunk 2]
```diff
@@ -1,82 +1,57 @@
...
 import javax.sql.DataSource;
 import java.io.Serializable;
+import java.util.Optional;
 
-import static dev.enginecode.eccommons.infrastructure.json.errors.InfrastructureErrorCode.*;
+import static dev.enginecode.eccommons.infrastructure.json.errors.InfrastructureErrorCode.RESOURCE_NOT_FOUND;
+import static dev.enginecode.eccommons.infrastructure.json.errors.InfrastructureErrorCode.TABLENAME_ANNOTATION_EMPTY;
 
 @Repository
 public class JsonMappingRepository<ID extends Serializable> implements JsonRepository<ID> {
```
Comment (tom-zoladz) [inline src/main/java/dev/enginecode/eccommons/infrastructure/json/repository/JsonMappingRepository.java:22]: Now, following the naming convention `PostgresDatabaseConnection` introduced, I think that this class could also have name specific for the particular database system like `JsonPostgresRepository`, what do you think?

**Review by tom-zoladz — COMMENTED** [review]
Generally I like the approach, some small comments added. Could you also add the `save(...)` method and serialising tests from `feature/save-record-to-db`?

File: src/main/java/dev/enginecode/eccommons/infrastructure/json/repository/mapping/Mapping.java (hunk 1 above)
Comment (michalrys) [reply src/main/java/dev/enginecode/eccommons/infrastructure/json/repository/mapping/Mapping.java]: OK, renamed to ObjectMapping - the main purpose of it is to map object to sth else by reading or writing methods.

File: src/main/java/dev/enginecode/eccommons/infrastructure/json/repository/mapping/JSONMapper.java
[the remark this answers is not in the document]
Comment (michalrys) [reply src/main/java/dev/enginecode/eccommons/infrastructure/json/repository/mapping/JSONMapper.java:1]: renamed to read method - it is more intuitive in my opinion, keep the sens what is going on. The same method has jackson ObjectMapper. Analogical will be write().

File: src/main/java/dev/enginecode/eccommons/infrastructure/json/repository/JsonMappingRepository.java (hunk 2 above)
Comment (michalrys) [reply src/main/java/dev/enginecode/eccommons/infrastructure/json/repository/JsonMappingRepository.java]: OK, renamed.

**Comment by michalrys** [comment]
> Generally I like the approach, some small comments added. Could you also add the `save(...)` method and serialising tests from `feature/save-record-to-db`?

no, because adding write method is in dedicated ticket: https://github.com/EngineCodeDev/weeds-plantcare-service/issues/26
This ticket is only for refactoring. Lest close it and then implement new method and new unit test related to it.

**Review by tom-zoladz — APPROVED** [review]

**Pull request merged**